cpp/src/toppra/geometric_path/piecewise_poly_path.cpp

```diff
@@ -152,7 +153,7 @@ size_t PiecewisePolyPath::findSegmentIndex(value_type pos) const {
   }
   auto it = std::upper_bound(m_breakpoints.begin(), m_breakpoints.end(), pos);
   auto idx = std::distance(m_breakpoints.begin(), it)-1;
-  return std::min(static_cast<size_t>(std::max(idx, long{0})), m_coefficients.size() - 1);
+  return std::min<size_t>(std::max<size_t>(idx, 0), m_coefficients.size() - 1);
```

Comment:
This one I'm not so sure @hungpham2511

Reply:
If the tests are passing I suppose it's fine. @jmirabel would you want to have a look?

Reply:
Make `idx` a `std::size_t` rather than `auto`. Then, I think on Windows, you have to put `std::min` and `std::max` between parenthesis to avoid min and max being treated as macros (not 100% sure I remember correctly the reason.

Then, this should do the work.
```suggestion
  return (std::min)((std::max)(idx, std::size_t(0)), m_coefficients.size() - 1);
```

Reply:
The `long{0}` confuses thecompiler if `idx` is `std::size_t`. Im concerned about some potential overflows when dealing with signed /unsigned.